src/main/java/org/folio/processing/mapping/mapper/reader/record/marc/MarcRecordReader.java

```diff
@@ -214,8 +227,41 @@ private void processStringExpression(MappingRule ruleExpression, boolean arrayVa
   }
 
   private void processTodayExpression(StringBuilder sb) {
-    SimpleDateFormat df = new SimpleDateFormat(ISO_DATE_FORMAT);
-    sb.append(df.format(new Date()));
+    try {
+      String mappingParams = eventPayload.getContext().get(MAPPING_PARAMS);
+      if (StringUtils.isNotEmpty(mappingParams)) {
+        MappingParameters mappingParameters = new JsonObject(mappingParams).mapTo(MappingParameters.class);
+        String tenantConfiguration = mappingParameters.getTenantConfiguration();
+        String tenantTimezone;
+        if (ifTimezonePatameterIsValid(tenantConfiguration)) {
+          tenantTimezone = UTC_TIMEZONE; // default if timezone configuration is empty.
+        } else {
+          tenantTimezone = new JsonObject(tenantConfiguration).getString(TIMEZONE_PROPERTY);
+        }
+        String currentDateTime = new SimpleDateFormat(DATE_TIME_FORMAT).format(new Date());
+        LocalDateTime ldt = LocalDateTime.parse(currentDateTime, DateTimeFormatter.ofPattern(DATE_TIME_FORMAT));
+        ZoneId utcZoneId = ZoneId.of(UTC_TIMEZONE);
+        ZonedDateTime utcZonedDateTime = ldt.atZone(utcZoneId);
+
+        ZoneId currentZoneId = ZoneId.of(tenantTimezone);
+        ZonedDateTime currentZonedDateTime = utcZonedDateTime.withZoneSameInstant(currentZoneId); //convert time from UTC to tenant's timezone(if exists).
+
+        DateTimeFormatter format = DateTimeFormatter.ofPattern(ISO_DATE_FORMAT);
+        sb.append(format.format(currentZonedDateTime));
+      } else {
+        SimpleDateFormat df = new SimpleDateFormat(ISO_DATE_FORMAT);
+        sb.append(df.format(new Date()));
```

Comment:
Better to use LocalDate, because Date has disadvantages, for example here listed some of them https://javarevisited.blogspot.com/2017/04/5-reasons-why-javas-old-date-and-Calendar-API-bad.html#axzz726MHMLZb

Reply:
@SerhiiNosko Thank you for this link. Changed it. Could you please double-check it?